swap/swap.go

```diff
@@ -305,16 +305,20 @@ func (s *Swap) handleEmitChequeMsg(ctx context.Context, p *Peer, msg *EmitCheque
 		return err
 	}
 
-	opts := bind.NewKeyedTransactor(s.owner.privateKey)
-	opts.Context = ctx
-
 	otherSwap, err := contract.InstanceAt(cheque.Contract, s.backend)
 	if err != nil {
 		return err
 	}
 
-	// cash cheque in async, otherwise this blocks here until the TX is mined
-	go defaultCashCheque(s, otherSwap, opts, cheque)
+	paidOut, err := otherSwap.PaidOut(nil, cheque.Beneficiary)
+	gasPrice := 30000000000 // 30 Gwei
+	// do a payout transaction if we get 2 times the gas costs
+	if (cheque.CumulativePayout - paidOut.Uint64()) > uint64(2*gasPrice) {
```

Comment:
with these prices this is always true, right? Every cheque is for 21 080 GWei. I'm not arguing that condition like this should be here, but maybe it should be set higher.

Reply:
Ah, it must be gasPrice times transactionCosts of course

Reply:
Can you confirm that it is better now? The idea is *not* to directly cash out

Reply:
It's cleaner now.

Not a problem, but still, based on the cheque values, it looks like it will most likely just be cashed right away unless gas cost increase drastically.